### PR TITLE
move to java 11 for groovy

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

File: sbin/build.sh [hunk 1]
```diff
@@ -371,14 +371,36 @@ executeTemplatedFile() {
...
+  echo $gradleJavaHome
+}
+
 buildSharedLibs() {
     cd "${LIB_DIR}"
-    GRADLE_USER_HOME=./gradle-cache ./gradlew clean uberjar
+    local gradleJavaHome=$(getGradleHome)
+    echo "Running gradle with $gradleJavaHome"
+
+    JAVA_HOME="$gradleJavaHome" GRADLE_USER_HOME=./gradle-cache ./gradlew clean uberjar
```
Comment (karianna) [inline sbin/build.sh]: Split into separate lines please

File: sbin/build.sh [hunk 2]
```diff
@@ -371,14 +371,36 @@ executeTemplatedFile() {
...
+
+    # Test parser
+    "$gradleJavaHome"/bin/java -version 2>&1 | "$gradleJavaHome"/bin/java -cp "${LIB_DIR}/target/libs/adopt-shared-lib.jar" ParseVersion -s -f semver 1
```
Comment (karianna) [inline sbin/build.sh]: Could be clearer as to what's this is testing in the comment, e.g. What's the expected output

File: sbin/common/common.sh [hunk 3]
```diff
@@ -33,6 +33,24 @@ function setOpenJdkVersion() {
...
+function crossPlatformRealPath() {
+  local target=$1
+
+  local currentDir="$PWD"
+
+  if [[ -d $target ]]; then
+    cd "$target"
+    local name=""
+  elif [[ -f $target ]]; then
+    cd "$(dirname "$target")"
```
Comment (karianna) [inline sbin/common/common.sh]: Is 'double' double quoting like this legal?

File: sbin/build.sh (hunk 1 above)
Comment (johnoliver) [reply sbin/build.sh]: you cant really

**Pull request merged**